kfbot-refactor-phase2/src/main/java/com/twitchbotx/bot/handlers/CountHandler.java

```diff
@@ -141,31 +141,63 @@ public String totals() {
-        public enum SubType{MASSGIFT, SINGLEGIFT, NEWPRIMESUB, NEWSUB, PRIMERESUB, RESUB}
-
-        // TODO: The returned value should be one of the parameters passed in to handleSubMessage
-        private static SubType assessSubType(boolean massGift, boolean singleGift, String subMonths, boolean prime) {
-            if (massGift) {
-                return SubType.MASSGIFT;
-            }
-
-            if (singleGift) {
-                return SubType.SINGLEGIFT;
-            }
-
-            if (Integer.parseInt(subMonths) < 2 && prime) {
-                return SubType.NEWPRIMESUB;
-            }
-
-            if (Integer.parseInt(subMonths) < 2) {
-                return SubType.NEWSUB;
-            }
-
-            if (prime) {
-                return SubType.PRIMERESUB;
+        public enum SubType {
+            MASSGIFT {
+                @Override
+                public String getTemplate() {
+                    return guiHandler.bot.getStore().getConfiguration().subMassGiftReply;
+                }
+            },
+            SINGLEGIFT {
+                @Override
+                public String getTemplate() {
+                    return guiHandler.bot.getStore().getConfiguration().subSingleGiftReply;
+                }
+            },
+            NEWPRIMESUB {
+                @Override
+                public String getTemplate() {
+                    return guiHandler.bot.getStore().getConfiguration().subNewPrimeReply;
+                }
+            },
+            NEWSUB {
+                @Override
+                public String getTemplate() {
+                    return guiHandler.bot.getStore().getConfiguration().subNewNormalReply;
+                }
+            },
+            PRIMERESUB {
+                @Override
+                public String getTemplate() {
+                    return guiHandler.bot.getStore().getConfiguration().subPrimeReply;
+                }
+            },
+            RESUB {
+                @Override
+                public String getTemplate() {
+                    return guiHandler.bot.getStore().getConfiguration().subNormalReply;
+                }
+            };
+
+            public String buildMessage(
+                String subDisplayName,
+                String tier,
+                String subMonths,
+                String giftRecipient,
+                int giftAmount,
+            ) {
+                String message = this.getTemplate()
+                
+                message = message.replace("%user", subDisplayName)
+                message = message.replace("%tier", tier)
+                message = message.replace("%months", subMonths)
+                message = message.replace("%recipient", giftRecipient)
+                message = message.replace("%gifts", String.valueOf(giftAmount))
+
+                return message
             }
 
-            return SubType.RESUB;
+            public abstract void getTemplate()
```

Comment:
; after this

Reply:
this method needs to also return a String instead of void

Reply:
woops